spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -1458,7 +1488,13 @@ protected void doInvokeListener(ChannelAwareMessageListener listener, Channel ch
 			}
 			// Actually invoke the message listener...
 			try {
-				listener.onMessage(message, channelToUse);
+				if (data instanceof List) {
+					((ChannelAwareBatchMessagelistener) listener).onMessageBatch((List<Message>) data, channelToUse);
```

Comment:
We don't need to cast here: The `ChannelAwareMessageListener` has that `onMessageBatch()` method